lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt

```diff
@@ -11,12 +11,28 @@ class KonigKontext<KontextType>(
     internal val valueToBinary: (KontextType) -> ByteArray,
     internal val valueFromBinary: (ByteArray) -> KontextType,
 ) {
+    init {
+        // Improper implementations of valueToBinary and valueFromBinary could potentially cause runtime exceptions without proper testing.
+        // This requirement should prevent developers from shooting themselves in the foot by eliminating that possibility.
+        require(
+            defaultValue == valueFromBinary(valueToBinary(defaultValue))
+        ) { "Unable to initialize KonigKontext. The valueFromBinary and valueToBinary functions do not seem to be implemented properly for the default value: $defaultValue" }
+    }
+
     internal val grpcContextKey = Context.key<KontextType>("konig-kontext-grpc-context")
     internal val grpcHeaderKey = KONIG_KONTEXT_GRPC_HEADER_KEY
 
     fun get(): KontextType = grpcContextKey.get() ?: defaultValue
 
-    fun setValue(konigKontext: KontextType) = KonigKontextRunnable(konigKontext)
+    fun setValue(konigKontext: KontextType) = run {
+        // Improper implementations of valueToBinary and valueFromBinary could potentially cause runtime exceptions without proper testing.
+        // This requirement should prevent developers from shooting themselves in the foot by eliminating that possibility.
+        require(
+            konigKontext == valueFromBinary(valueToBinary(konigKontext))
+        ) { "Unable to set KonigKontext. The valueFromBinary and valueToBinary functions do not seem to be implemented properly for value: $konigKontext" }
+
+        KonigKontextRunnable(konigKontext)
+    }
```

Comment:
Not super sure about these additions yet. They will enforce that `valueToBinary` and `valueFromBinary` are implemented properly at runtime, so presumably any issues would get caught before reaching production if there is any sort of testing in place. However, they could impact performance slightly as every RPC call will require parsing the konig kontext value to/from binary format. Thoughts?

Reply:
hm could you wrap it in something that gets compiled away in production builds?

Reply:
Maybe.. Will explore this further in other changes. I removed these checks for now, and we will currently throw a helpful error message at runtime if either valueToBinary or valueFromBinary is not implemented properly during a request